### PR TITLE
[SCAL-83090] left panel collapse

### PR DESCRIPTION
**Pull request by harsh-ts**
Ticket Link: https://thoughtspot.atlassian.net/browse/SCAL-83090

### CODE REVIEW

File: docs/src/components/LeftSidebar/helper.ts [hunk 1]
```diff
@@ -16,8 +16,13 @@ export const addExpandCollapseImages = (navContent: string, pageId: string) => {
             const spanElement = document.createElement('span');
             spanElement.classList.add('iconSpan');
             const imageElement = document.createElement('img');
-            imageElement.src = ArrowForward;
-            divElement.classList.add('displayNone');
+            const tabIndex = tabsClosed.findIndex(i => i === index);
+            if (tabIndex >= 0) {
```
Comment (jbhanu-thoughtspot) [inline docs/src/components/LeftSidebar/helper.ts:20]: tabsClosed.includes(index)

File: docs/src/components/LeftSidebar/index.tsx [hunk 2]
```diff
@@ -55,14 +61,29 @@ const LeftSideBar = (props: {
...
+        const tabsArr = [...tabsRef.current];
+        const tabIndex = tabsArr.findIndex((i) => i === index);
+        if (tabIndex >= 0) {
+            tabsArr.splice(tabIndex, 1);
+        } else {
+            tabsArr.push(index);
+        }
+        tabsRef.current = [...tabsArr];
+    };
+
```
Comment (jbhanu-thoughtspot) [inline docs/src/components/LeftSidebar/index.tsx:86]:
```
const handleTabClick = (tabIndex: number) => {
    tabsRef.current = tabsRef.current.includes(tabIndex) ?
        tabsRef.current.filter(idx => idx !== tabIndex) :
        [...tabsRef.current, tabIndex];
};
```

**Review by jbhanu-thoughtspot — CHANGES_REQUESTED** [review]

File: docs/src/components/LeftSidebar/helper.ts (hunk 1 above)
Comment (harsh-ts) [reply docs/src/components/LeftSidebar/helper.ts:20]: Will make suggested changes

File: docs/src/components/LeftSidebar/index.tsx (hunk 2 above)
Comment (harsh-ts) [reply docs/src/components/LeftSidebar/index.tsx:86]: Will make suggested changes

File: docs/src/components/LeftSidebar/index.tsx [hunk 3]
```diff
@@ -34,6 +34,8 @@ const LeftSideBar = (props: {
     const [navContent, setNavContent] = useState('');
     const { width, ref, height } = useResizeDetector();
 
+    const tabsRef = useRef([]);
```
Comment (jbhanu-thoughtspot) [inline docs/src/components/LeftSidebar/index.tsx:37]: Let's name this `expandedTabsRef` to give better context on the use of this variable

File: docs/src/components/LeftSidebar/index.tsx [hunk 4]
```diff
@@ -55,14 +61,24 @@ const LeftSideBar = (props: {
...
+            handleTabClick,
+        );
     }, [params[TS_PAGE_ID_PARAM], isMaxMobileResolution, navContent]);
 
     const onMenuClick = () => {
         props.setLeftNavOpen(!props.leftNavOpen);
         document.documentElement.scrollTop = 0;
     };
 
+    const handleTabClick = (tabIndex: number) => {
```
Comment (jbhanu-thoughtspot) [inline docs/src/components/LeftSidebar/index.tsx:76]: Name this method toggleExpandOnTab

**Review by jbhanu-thoughtspot — APPROVED** [review]

**Pull request merged**